### PR TITLE
Additional Stargate Related Doc Updates #139

### PR DESCRIPTION
**Pull request by jdonenine**
* Initial pass at updating docs with additional relevant info regarding the Stargate integration
* Also includes a little rework to make it easier to reference the value files in the docs using some symbolic links that can then be dynamically loaded (instead of copying and pasting over time)
* Once we add in the out-of-the-box ingress support we'll want to come back and update the ingress portions of the doc based on that.

### CODE REVIEW

File: docs/content/en/docs/getting-started/_index.md [hunk 1]
```diff
@@ -42,6 +42,25 @@ helm repo update
 
 Alternatively, you may download the individual charts directly from the project's [releases](https://github.com/k8ssandra/k8ssandra/releases) page.
 
+## Enabling Stargate
+
+Stargate deployment is *disabled* by default, to enable it, the values provided to the helm install/upgrade of the
+`k8ssandra-cluster` chart can be modified.
```
Comment (jakerobb) [inline docs/content/en/docs/getting-started/_index.md:48]:
```suggestion
Stargate deployment is *disabled* by default. To enable it, modify the values provided to the helm install/upgrade of the
`k8ssandra-cluster` chart.
```

File: docs/content/en/docs/getting-started/_index.md [hunk 2]
```diff
@@ -42,6 +42,25 @@ helm repo update
...
+
+This can be done by adding to the values.yaml file provided to helm:
+
+```
+stargate:
+  enabled: true
+  replicas: 1
```
Comment (jakerobb) [inline docs/content/en/docs/getting-started/_index.md:55]: `replicas` is set to 1 by default. I would omit this here; I don't think it's a good idea to encourage redundant configuration.

File: docs/content/en/docs/getting-started/_index.md [hunk 3]
```diff
@@ -42,6 +42,25 @@ helm repo update
...
+```
+
+or set via the command line when installing via helm:
+
+```
+--set stargate.enabled=true --set stargate.replicas=1
```
Comment (jakerobb) [inline docs/content/en/docs/getting-started/_index.md:61]: remove `replicas` here as well

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md [hunk 4]
```diff
@@ -6,39 +6,141 @@ description: |
   Accessing the K8ssandra Stargate interfaces.
 ---
 
-[Stargate](https://stargate.io/) is an open-source framework providing common
+[Stargate](https://stargate.io/) is an open source data gateway providing common
```
Comment (jakerobb) [inline docs/content/en/docs/topics/accessing-services/stargate/_index.md:9]: "Open-source" should be hyphenated when used as an adjective.

```suggestion
[Stargate](https://stargate.io/) is an open-source data gateway providing common
```

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md [hunk 5]
```diff
@@ -6,39 +6,141 @@ description: |
...
 API interfaces for backend databases. With K8ssandra, Stargate may be deployed
-in front of the Apache Cassandra cluster providing CQL, REST, and GraphQL API
-endpoints. These endpoints may be scaled horizontally independently of data
-layer scaling needs. This guide covers accessing the various API endpoints
+in front of the Apache Cassandra cluster providing CQL, REST, GraphQL, and document based API
```
Comment (jakerobb) [inline docs/content/en/docs/topics/accessing-services/stargate/_index.md:11]: More hyphenation :)

```suggestion
in front of the Apache Cassandra cluster providing CQL, REST, GraphQL, and document-based API
```

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md [hunk 6]
```diff
@@ -6,39 +6,141 @@ description: |
...
+endpoints. These endpoints may be scaled independently of the data
```
Comment (jakerobb) [inline docs/content/en/docs/topics/accessing-services/stargate/_index.md:12]: The double adverb was awkward, but I think "horizontally" was contributing useful value to the sentence -- we're talking about increasing the number of Stargate endpoints, not increasing the resources allocated to each endpoint. I don't have a good suggestion offhand for a rephrase, but let's spend a minute and see if we can work that tidbit back in here.

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md [hunk 7]
```diff
@@ -6,39 +6,141 @@ description: |
...
-   DOMAIN_ below.
-1. Port number for the Stargate CQL interface
+1. [K8ssandra Cluster]({{< ref "getting-started#install-k8ssandra" >}})
+1. [Ingress]({{< ref "ingress" >}}) configured to expose each of the Stargate services (Auth, REST, GraphQL)
+1. DNS names configured for the exposed Stargate services, referred to as `STARGATE_AUTH_DOMAIN`, `STARGATE_REST_DOMAIN`, and `STARGATE_GRAPHQL_DOMAIN` below.
+
+## Access Auth API
+
+Before accessing any of the provided Stargate data APIs, an auth token must be generated and provided
+to subsequent data API requests.  This token generation process is done via the auth API.
```
Comment (jakerobb) [inline docs/content/en/docs/topics/accessing-services/stargate/_index.md:34]:
```suggestion
to subsequent data API requests.  Use the auth API to generate a token.
```

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md [hunk 8]
```diff
@@ -6,39 +6,141 @@ description: |
...
+
+Extract and decode the password secret:
+
+```
+kubectl get secret k8ssandra-cluster-superuser -o jsonpath="{.data.password}" | base64 --decode
+```
+
+### Generating Auth Tokens
+
+With the secrets extract and decoded, a request to generate a token can now be sent to the Stargate auth API.
```
Comment (jakerobb) [inline docs/content/en/docs/topics/accessing-services/stargate/_index.md:63]: Grammar nitpick:

```suggestion
With the secrets extracted and decoded, a request to generate a token can now be sent to the Stargate auth API.
```

Also, the second half of the sentence is passive voice. I'd be tempted to rephrase as something like "we are now ready to request a token from the Stargate auth API," but the use of "we" might go against style guidelines. Do we have any such guidelines?

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md [hunk 9]
```diff
@@ -6,39 +6,141 @@ description: |
...
+
+```
+curl -L -X POST 'http://_STARGATE_DOMAIN_/v1/auth' -H 'Content-Type: application/json' --data-raw '{"username": "k8ssandra-cluster-superuser", "password": "1LI8TebjjHYrqUk9xYbJnbYJheX3Ckq250byd2ePDPXNtweaYgznmg"}'
+```
+
+This request will return a response similar to below providing the token that will be required when making 
+requests to the Stargate data APIs.
```
Comment (jakerobb) [inline docs/content/en/docs/topics/accessing-services/stargate/_index.md:70]: I had to read this sentence three times before it parsed correctly in my head.

```suggestion
This request will return a response similar to the following. The value given for `authToken` will be required when making requests to the Stargate data APIs.
```

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md [hunk 10]
```diff
@@ -6,39 +6,141 @@ description: |
...
+
+```
+{"authToken":"e4b34bbc-0ebc-4e2a-86ca-04793ca658a7"}
+```
+
+### Using Auth Tokens
+
+Stargate supports authorization within the data APIs through a custom header `x-cassandra-token` which must 
+be populated with the token produced through the use of the auth API previously described.
```
Comment (jakerobb) [inline docs/content/en/docs/topics/accessing-services/stargate/_index.md:79]:
```suggestion
Stargate supports authorization within the data APIs through a custom HTTP header `x-cassandra-token`, which must be populated with the token given by the auth API.
```

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md (hunk 5 above)
Comment (jakerobb) [reply docs/content/en/docs/topics/accessing-services/stargate/_index.md:11]: or:
```suggestion
in front of the Apache Cassandra cluster providing CQL, REST, GraphQL, and document API
```

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md [hunk 11]
```diff
@@ -6,39 +6,141 @@ description: |
...
+
+## Access Document Data API
+
+The Stargate document APIs provide a way schemaless way to store and interact with data inside of Cassandra.
+The first step to using the document API is to [create a namespace](https://stargate.io/docs/stargate/1.0/quickstart/quick_start-document.html#_creating_schema).
```
Comment (jakerobb) [inline docs/content/en/docs/topics/accessing-services/stargate/_index.md:84]:
```suggestion
The first step is to [create a namespace](https://stargate.io/docs/stargate/1.0/quickstart/quick_start-document.html#_creating_schema). 
```

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md [hunk 12]
```diff
@@ -6,39 +6,141 @@ description: |
...
+{"name":"mynamespace"}
+```
+
+Additional information related to using the Document APIs can be found in the Stargate [docs](https://stargate.io/docs/stargate/1.0/quickstart/quick_start-document.html).
+
+## Access REST Data API
+
+The Stargate REST APIs provide a RESTful way to store and interact with data inside of Cassandra that should feel
+familiar to developers, but unlike the document APIs, does require some additional Cassandra data modeling 
+understanding. The first step to using the REST APIs is to [create a keyspace](https://stargate.io/docs/stargate/1.0/quickstart/quick_start-rest.html#_creating_schema).
```
Comment (jakerobb) [inline docs/content/en/docs/topics/accessing-services/stargate/_index.md:109]:
```suggestion
familiar to developers. Unlike the document APIs, some understanding of Cassandra data modeling will be required. The first step is to [create a keyspace](https://stargate.io/docs/stargate/1.0/quickstart/quick_start-rest.html#_creating_schema). 
```

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md [hunk 13]
```diff
@@ -6,39 +6,141 @@ description: |
...
+Additional information related to using the Document APIs can be found in the Stargate [docs](https://stargate.io/docs/stargate/1.0/quickstart/quick_start-rest.html).
+
+## Access GraphQL Data API
+
+The Stargate GraphQL APIs provide a way to store and interact with data inside of Cassandra using the powerful GraphQL
+query language and tooling ecosystem. Like the REST APIs, this does require some additional Cassandra data modeling
+understanding. Like the REST APIs, The first step to using the GraphQL APIs is to [create a keyspace](https://stargate.io/docs/stargate/1.0/quickstart/quick_start-graphql.html#_creating_schema).
 
-## Access REST interface
+The easiest way to get started with the GraphQL APIs is to use the built-in GraphQL playground, described below.
```
Comment (jakerobb) [inline docs/content/en/docs/topics/accessing-services/stargate/_index.md:136]:
```suggestion
The easiest way to get started with the GraphQL APIs is to use the built-in GraphQL playground described in the next section.
```

**Review by jakerobb — CHANGES_REQUESTED** [review]
Nice! I added a handful of grammatical nitpicks and wording improvements.

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md (hunk 8 above)
Comment (jakerobb) [reply docs/content/en/docs/topics/accessing-services/stargate/_index.md:63]: Perhaps:
```suggestion
Use the extracted and decoded secrets to request a token from the Stargate auth API.
```

File: docs/content/en/docs/getting-started/_index.md (hunk 2 above)
Comment (jdonenine) [reply docs/content/en/docs/getting-started/_index.md:55]: Make sense.  Done.

File: docs/content/en/docs/getting-started/_index.md (hunk 3 above)
Comment (jdonenine) [reply docs/content/en/docs/getting-started/_index.md:61]: Makes sense.  Done.

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md (hunk 6 above)
Comment (jdonenine) [reply docs/content/en/docs/topics/accessing-services/stargate/_index.md:12]: How about `Stargate itself may be scaled horizontally within the cluster as needed.  This scaling is done independently of the data layer scaling.`?  With a link to the section for scaling the data layer?

**Comment by jdonenine** [comment]
Thanks for the review @jakerobb I pulled in most all of your suggestions and added a potential thought to the one remaining question.  Let me know what you think.

**Review by jsanda — COMMENTED** [review]
Do we have documented the versions of Cassandra that are supported?

File: docs/content/en/docs/topics/accessing-services/stargate/_index.md (hunk 6 above)
Comment (jdonenine) [reply docs/content/en/docs/topics/accessing-services/stargate/_index.md]: Updated this in the last commit to bring back the note about the independent scaling.

**Comment by jdonenine** [comment]
> Do we have documented the versions of Cassandra that are supported?

That's a good question... I don't think we do.  We'll definitely want to add that in the 1.0 doc effort as we move towards that.  I'll create an issue to cover that need.

**Comment by jdonenine** [comment]
> > Do we have documented the versions of Cassandra that are supported?
> 
> That's a good question... I don't think we do. We'll definitely want to add that in the 1.0 doc effort as we move towards that. I'll create an issue to cover that need.

Created #146 to cover this.

**Review by jsanda — APPROVED** [review]

**Review by jakerobb — APPROVED** [review]
Looks good to me. :)

**Review by jsanda — APPROVED** [review]

**Pull request merged**